massa-grpc/src/tests/mock.rs

```diff
@@ -36,192 +26,20 @@ use massa_versioning::versioning::{MipStatsConfig, MipStore};
-        fn get_block_endorsements(
-            &self,
-            target_block: &BlockId,
-            slot: &Slot,
-        ) -> (Vec<Option<EndorsementId>>, Storage);
-
-        /// Get denunciations for a block header.
-        fn get_block_denunciations(&self, target_slot: &Slot) -> Vec<Denunciation>;
-
-        /// Get the number of endorsements in the pool
-        fn get_endorsement_count(&self) -> usize;
-
-        /// Get the number of operations in the pool
-        fn get_operation_count(&self) -> usize;
-
-        /// Check if the pool contains a list of endorsements. Returns one boolean per item.
-        fn contains_endorsements(&self, endorsements: &[EndorsementId]) -> Vec<bool>;
-
-        /// Check if the pool contains a list of operations. Returns one boolean per item.
-        fn contains_operations(&self, operations: &[OperationId]) -> Vec<bool>;
-
-        /// Check if the pool contains a denunciation. Returns a boolean
-        fn contains_denunciation(&self, denunciation: &Denunciation) -> bool;
-
-        /// Get the number of denunciations in the pool
-        fn get_denunciation_count(&self) -> usize;
-
-        /// Returns a boxed clone of self.
-        /// Useful to allow cloning `Box<dyn PoolController>`.
-        fn clone_box(&self) -> Box<dyn PoolController>;
-
-        /// Get final cs periods (updated regularly from consensus)
-        fn get_final_cs_periods(&self) -> &Vec<u64>;
-    }
-}
-
-mockall::mock! {
-    pub SelectorCtrl {}
-
-    impl SelectorController for SelectorCtrl {
-        fn wait_for_draws(&self, cycle: u64) -> PosResult<u64>;
-
-        fn feed_cycle(
-            &self,
-            cycle: u64,
-            lookback_rolls: BTreeMap<Address, u64>,
-            lookback_seed: massa_hash::Hash,
-        ) -> PosResult<()>;
-
-        fn get_selection(&self, slot: Slot) -> PosResult<Selection>;
-
-        fn get_producer(&self, slot: Slot) -> PosResult<Address>;
-
-        #[allow(clippy::needless_lifetimes)] // lifetime elision conflicts with Mockall
-        fn get_available_selections_in_range<'a>(
-            &self,
-            slot_range: std::ops::RangeInclusive<Slot>,
-            restrict_to_addresses: Option<&'a PreHashSet<Address>>,
-        ) -> PosResult<BTreeMap<Slot, Selection>>;
-
-        fn clone_box(&self) -> Box<dyn SelectorController>;
-
-        #[cfg(feature = "testing")]
-        fn get_entire_selection(&self) -> std::collections::VecDeque<(u64,HashMap<Slot,Selection>)> {
-            unimplemented!("mock implementation only")
-        }
-    }
-}
-
 /// generate a grpc public service
 /// # Arguments
 /// * `addr` - the address to bind to
 /// # Returns
 /// * `MassaPublicGrpc` - the grpc public service
 #[allow(dead_code)]
 pub(crate) fn grpc_public_service(addr: &SocketAddr) -> MassaPublicGrpc {
-    let consensus_controller = MockConsensusControllerImpl::new();
-
+    let consensus_ctrl = Box::new(MockConsensusController::new());
```

Comment:
What's the goal of this deadcode function? Should we put it somewhere else now that "mock.rs" contains only that?

Reply:
It comes from an other PR but it seems that a confusion was made between testing and test and so I fixed it which remove the utility of this.